components/new_post/post_component.tsx

```diff
@@ -0,0 +1,498 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {MouseEvent, useCallback, useEffect, useRef, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import classNames from 'classnames';
+
+import {Posts} from 'mattermost-redux/constants/index';
+import {
+    isMeMessage as checkIsMeMessage,
+    isPostPendingOrFailed,
+} from 'mattermost-redux/utils/post_utils';
+
+import Constants, {A11yCustomEventTypes, AppEvents, Locations} from 'utils/constants';
+import * as PostUtils from 'utils/post_utils';
+
+import {Post} from '@mattermost/types/posts';
+import {Emoji} from '@mattermost/types/emojis';
+import {PostPluginComponent} from 'types/store/plugins';
+
+import FileAttachmentListContainer from 'components/file_attachment_list';
+import DateSeparator from 'components/post_view/date_separator';
+import OverlayTrigger from 'components/overlay_trigger';
+import Tooltip from 'components/tooltip';
+import PostProfilePicture from 'components/post_profile_picture';
+import FailedPostOptions from 'components/post_view/failed_post_options';
+import PostAriaLabelDiv from 'components/post_view/post_aria_label_div';
+import PostTime from 'components/post_view/post_time';
+import ReactionList from 'components/post_view/reaction_list';
+import MessageWithAdditionalContent from 'components/message_with_additional_content';
+import InfoSmallIcon from 'components/widgets/icons/info_small_icon';
+import ArchiveIcon from 'components/widgets/icons/archive_icon';
+import PostPreHeader from 'components/post_view/post_pre_header';
+import EditPost from 'components/edit_post';
+import AutoHeightSwitcher, {AutoHeightSlots} from 'components/common/auto_height_switcher';
+import {Props as TimestampProps} from 'components/timestamp/timestamp';
+import ThreadFooter from 'components/threading/channel_threads/thread_footer';
+import PostBodyAdditionalContent from 'components/post_view/post_body_additional_content';
+import PostMessageContainer from 'components/post_view/post_message_view';
+import * as Utils from 'utils/utils';
+import {getHistory} from 'utils/browser_history';
+
+import PostUserProfile from './user_profile';
+import PostOptions from './post_options';
+
+export type Props = {
+    post: Post;
+    teamId: string;
```

Comment:
this is redundant since we already defined OwnProps in index file. and the redux props can be taken from connector. Check out components/channel_notifications_modal/index.ts

Reply:
but this way we are doing circular imports tho, is this what we want?

Reply:
it wont be because we do import type only
```js
import type {Foo} from 'xx'
```
Which will be stripped out eventually